### PR TITLE
Enable passing current latlng as query params + some fixes

### PR DESCRIPTION
**Pull request by arjun27**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -105,17 +104,13 @@ class App extends Component {
...
   };
 
-  setGeolocation = location => {
-    this.setState({
-      geoLocation: location,
+  setCurrentLocation = ({ latLng, accuracy }) => {
+    return new Promise((resolve) => {
+      this.setState({
+        currentLocation: { latLng, accuracy }
+      }, resolve());
```
Comment (Siddharth11) [inline src/App.js:111]: shouldn't this be ?
```suggestion
      this.setState({
        currentLocation: { latLng, accuracy }
      }, resolve);
```

File: src/Homepage/index.js [hunk 2]
```diff
@@ -72,7 +71,6 @@ class Homepage extends React.Component {
   async fetchResults() {
     let selectedLocation;
```
Comment (Siddharth11) [inline src/Homepage/index.js:72]: You were right about confusing variable names. should we make it more descriptive, something like
`locationComingFromStoreSubmitCall`

File: src/Homepage/index.js [hunk 3]
```diff
@@ -132,10 +134,27 @@ class Homepage extends React.Component {
...
+        currentLocation = { latLng: currentLatLng, accuracy };
+      } catch (e) {
+        console.log('Cannot parse les query params.')
+      }
+    }
+
+    if (currentLocation) {
+      await this.props.setCurrentLocation(currentLocation);
+    }
+    await this.fetchResults();
```
Comment (Siddharth11) [inline src/Homepage/index.js:156]: let's add a try catch around this

File: src/App.js (hunk 1 above)
Comment (arjun27) [reply src/App.js:111]: oh wow, good catch

**Pull request merged**